### PR TITLE
[python][sklearn] respect `eval_at` aliases in keyword arguments

### LINKED ISSUE #4170: eval_at aliases do not change evaluation metrics
## Description
The various aliases of the `eval_at` parameter (used for `map` and `ndcg` metrics) does not change the "@" thresholds when calculating validation set metrics on `eval_set`. Only `eval_at` appears to work.

The `eval_at` aliases listed on the [Parameters docs page](https://lightgbm.readthedocs.io/en/latest/Parameters.html#eval_at) make it seem like users can also pass in
- `ndcg_eval_at`
- `ndcg_at`
- `map_eval_at`
- `map_at`

as valid threshold specifications. But when passed as a `.fit` keyword argument, LightGBM does not recognize these aliases as valid keywords.

Instead of passing these aliases to `.fit`, when they are passed in as estimator constructor arguments while using the sklearn API, they not have any impact the evaluation metric thresholds applied to `eval_set`. For example, I can set `LGBMRanker(ndcg_eval_at = [1, 5, 9])` but still only the default thresholds of `1,2,3,4,5` will be evaluated on any `eval_set` used during `.fit`.

Only `eval_at` as a keyword argument to `LGBMRanker.fit` appears to have any bearing on the final set of evaluation metrics.

I think there is confusion between treating `eval_at` as a `LGBMRanker.fit` kwarg and treating it as a ranker constructor kwarg. The default `eval_at` value on `LGBMRanker.fit` is `eval_at=(1,2,3,4,5)`. Then `eval_at` in `.fit` is used to override the `_eval_at` attribute of the ranker. See [here](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/sklearn.py#L982). So at the same time that there is no recognition during the part of `.fit` that any of the aliases of `eval_set` are valid aliases, the default `eval_at` setting is being used to override any possible settings the user may have provided for `eval_at` during estimator construction, thus eliminating any chance of ever using one of these aliases.

**Two proposed solutions**:
1. Drop support for these aliases. Already the `ndcg_at`, `ndcg_eval_at`, `map_at`, and `map_eval_at` aliases [all map to just `eval_at`](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/basic.py#L293). So if the user specifies `metric=["ndcg", "map"]`, it's not like `ndcg_at` **only** controls the NDCG threshold and `map_at` **only** controls the MAP threshold - `ndcg_at` controls MAP thresholds and vice versa. So these aliases can create confusion. Just use `eval_at` and drop these aliases from the docs page.
2. Remove `eval_at=(1,2,3,4,5)` as the explicit default in `LGBMRanker.fit`. Instead, make it `None`. During `LGBMRanker` construction, alias the `eval_at` aliases correctly and set the `_eval_at` attribute during `__init__`. If the user does not provide `eval_at` during `fit`, log a statement telling them that it will default to `(1,2,3,4,5)` only when no estimator-level attribute has been set during construction. But if the user has provided some `eval_at` alias to `LGBMRanker.__init__`, then use that setting by checking whether or not `_eval_at` already exists as an attribute on the ranker before defaulting to `(1,2,3,4,5)`.

## Reproducible example
```
import lightgbm as lgb
import numpy as np
import pandas as pd

# -- create training data, define eval_set evaluatioin thresholds.
X_df = pd.DataFrame(np.random.normal(size=(100,5)), columns=[f'f_{i}' for i in range(5)])
y_rnk = np.random.choice(range(4), size=100, replace=True)
grp = [10] * 10
eval_set = [(X_df.iloc[0:30, ], y_rnk[0:30])]
eval_grp = [grp[0:3]]
thresholds = [1, 5, 9]
thresholds_str = ','.join([str(x) for x in thresholds])
print(f'evaluation thresholds should be {thresholds}.')

# -- eval_at as a fit argument:
# -- this *does* change the eval_set evaluation thresholds.
rnk = lgb.LGBMRanker(n_estimators=10)
rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, eval_at=thresholds, verbose=0)
print(list(rnk.evals_result_['valid_0'].keys())) # 'ndcg@1', 'ndcg@5', 'ndcg@9'

# -- ncdg_eval_at as fit argument: throws execption "unexpected keyword argument 'ndcg_eval_at'"
rnk = lgb.LGBMRanker(n_estimators=10)
try:
	rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, ndcg_eval_at=thresholds, verbose=0)
except TypeError as e:
	print('ndcg_eval_at-as-fit-arg error: ', e)

# -- ncdg_eval_at as constructor arg: does not impact evaluation thresholds
rnk = lgb.LGBMRanker(n_estimators=10, metric='ndcg', ndcg_eval_at=thresholds)
rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, verbose=0)
print(list(rnk.evals_result_['valid_0'].keys())) # should be 'ndcg@1', 'ndcg@5', 'ndcg@9' but is @1-5

# -- ndcg_eval_at, but using string version of thresholds '1,5,9'.
# -- Similarly does not impact evaluation thresholds.
rnk = lgb.LGBMRanker(n_estimators=10, metric='ndcg', ndcg_eval_at=thresholds_str) 
rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, verbose=0)
print(list(rnk.evals_result_['valid_0'].keys())) # still ndcg@1-5

# -- map_at, map_eval_at similarly do not change eval_set thresholds.
rnk = lgb.LGBMRanker(n_estimators=10, metric='map')
try:
	rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, map_at=thresholds, verbose=0)
except TypeError as e:
	print('map_at-as-fit-arg error: ', e)

rnk = lgb.LGBMRanker(n_estimators=10, metric='map', map_at=thresholds_str)
rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, verbose=0)
print(list(rnk.evals_result_['valid_0'].keys())) # still map@1-5

rnk = lgb.LGBMRanker(n_estimators=10, metric='map', map_eval_at=thresholds_str)
rnk.fit(X_df, y_rnk, group=grp, eval_set=eval_set, eval_group=eval_grp, verbose=0)
print(list(rnk.evals_result_['valid_0'].keys())) # still map@1-5
```

> evaluation thresholds should be [1, 5, 9].
['ndcg@1', 'ndcg@5', 'ndcg@9']
ndcg_eval_at-as-fit-arg error:  fit() got an unexpected keyword argument 'ndcg_eval_at'
['ndcg@1', 'ndcg@2', 'ndcg@3', 'ndcg@4', 'ndcg@5']
['ndcg@1', 'ndcg@2', 'ndcg@3', 'ndcg@4', 'ndcg@5']
map_at-as-fit-arg error:  fit() got an unexpected keyword argument 'map_at'
['map@1', 'map@2', 'map@3', 'map@4', 'map@5']
['map@1', 'map@2', 'map@3', 'map@4', 'map@5']

## Environment info

LightGBM version or commit hash: 3.2.0

Command(s) you used to install LightGBM:
```shell
pip install lightgbm --upgrade
```

## Additional Comments
First mentioned in [this comment](https://github.com/microsoft/LightGBM/pull/4101#discussion_r601885587).

### PR DESCRIPTION
**Pull request by StrikerRUS**
Fixed #4170.

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]
nice change and excellent tests! Looks good.
